xdsl/dialects/pdl_interp.py

```diff
@@ -560,6 +584,105 @@ def __init__(
             },
         )
 
+    @classmethod
+    def parse(cls, parser: Parser) -> CreateOperationOp:
+        def _parse_attribute() -> tuple[StringAttr, SSAValue]:
+            attrname = parser.parse_attribute()
+            assert isinstance(attrname, StringAttr)
```

Comment:
similarly for the asserts in this function, either we should let the verification happen after the whole file is parsed, or we should raise a parse error